progress/README.md

```diff
@@ -0,0 +1,256 @@
+# Progress
+
+## **Requests**
+
+|                  Route                   |                         Description                          |
+| :--------------------------------------: | :----------------------------------------------------------: |
+|        [GET /progress](#get-post)        | Retrieves progress entries based on the provided parameters. |
+|     [POST /progress](#post-progress)     |                Creates a new progress entry.                 |
+| [DELETE /progress/:id](#delete-progress) |       Deletes a progress entry with the specified ID.        |
+
+## **GET /progress**
+
+Retrieves progress entries based on the provided parameters.
+
+- **Params**
+  None
+- **Query**  
+  type (query parameter): Specifies the type of progress to filter (e.g., "task", "user").
+  id (query parameter): Specifies the ID of the progress entry to retrieve (optional).
+- **Body**  
+  None
+- **Headers**  
+  Content-Type: application/json
+- **Cookie**  
+  None
+- **Success Response:**
+  - **Code:** 200
+    - **Content:** 
+    ```json
+		{
+      "message": "User progress retrieved successfully.",
+      "progress": [
+        { "<user_progress_object_1>" },
+        { "<user_progress_object_2>" },
+        ..
+      ]
+    }
+    ```
+- **Error Response:**
+  - **Code:** 500
+    - **Content:** `{ 'statusCode': 500, 'error': 'Internal Server Error', 'message': 'An internal server error occurred' }`
```

Comment:
internal server error for get? wont it be plain 404?

Reply:
sometimes we do get internal error for get if something goes wrong in the server side.